explorer/client/src/utils/stringUtils.ts

```diff
@@ -48,6 +50,47 @@ export function truncate(fullStr: string, strLen: number, separator?: string) {
     );
 }
 
+export async function extractFileType(
+    displayString: string,
+    signal: AbortSignal
+) {
+    let result: string;
+
+    try {
+        // First check Content-Type in header:
+        result = await fetch(transformURL(displayString), {
+            signal: signal,
+        })
+            .then(
+                (resp) =>
+                    resp?.headers
+                        ?.get('Content-Type')
+                        ?.split('/')
+                        .reverse()?.[0]
+                        ?.toUpperCase() || 'Image'
+            )
+            .catch((err) => {
+                console.error(err);
+                return 'Image';
+            });
+
+        // When Content-Type cannot be accessed (e.g. because of CORS), rely on file extension
+
+        if (result === 'Image') {
+            const extension =
+                displayString?.split('.').reverse()?.[0]?.toUpperCase() || '';
+            if (['JPG', 'JPEG', 'PNG'].includes(extension)) {
```

Comment:
why uppercase ? usually file extensions would be lower case.

Reply:
See the Figma design - file names are to be described in uppercase.